### PR TITLE
Client refactor

### LINKED ISSUE #11: Write Unit Tests

### LINKED ISSUE #12: Refactor code for testability
Refactoring to a more object-oriented approach with a Client object and clearly-delineated methods will allow for the code to actually be testable.

### LINKED ISSUE #13: Add Docker image publication to release workflow

### LINKED ISSUE #14: Create Helm chart to deploy the Docker container into a k8s cluster

### PR DESCRIPTION
**Pull request by tlake**
Changelog:

- Replaced `--region` runtime flag with `AWS_REGION` env var.
- Refactored codebase to use an `ECSClient` pattern.
- Updated `--debug` flag.
- Introduced automatic Docker image publishing to the CI/CD workflow.
- Added testing suite.
- Removed concurrency.
- Added Helm chart.

Closes #11 
Closes #12 
Closes #13 
Closes #14

### CODE REVIEW

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -74,4 +74,21 @@ jobs:
           upload_url: ${{ steps.create_release.outputs.upload_url }} 
           asset_path: ./build/Windows.zip
           asset_name: go-ecs-cleaner-windows.zip
-          asset_content_type: application/zip
\ No newline at end of file
+          asset_content_type: application/zip
+
+      - name: Publish Version to Registry
```
Comment (diemonster) [inline .github/workflows/release.yml:79]: this is awesome

File: cmd/ecs_task.go [hunk 2]
```diff
@@ -4,53 +4,70 @@ import (
...
+		ecsClient.Flags.Apply = applyFlag
+		ecsClient.Flags.Cutoff = cutoffFlag
+		ecsClient.Flags.Debug = debugFlag
+		ecsClient.Flags.Parallel = parallelFlag
+		ecsClient.Flags.Quiet = quietFlag
+		ecsClient.Flags.Verbose = verboseFlag
+
+		if err := ecsClient.CleanupTaskDefinitions(); err != nil {
+			fmt.Println(err)
+			os.Exit(1)
```
Comment (diemonster) [inline cmd/ecs_task.go:62]: Have you been able to confirm that the program cleanly exits on error here?

**Review by diemonster — CHANGES_REQUESTED** [review]
This looks great but without some tests to ensure the channel logic is working as intended, it seems like we're still stabbing in the dark re: catching and correctly dealing w errors.

**Comment by diemonster** [comment]
Just to sum up previous conversation: another option is to ditch concurrency all together and add a note in the README about scaling this "service" w docker. This would probably simplify the logic quite a bit, not to mention the testing.

**Comment by tlake** [comment]
Currently I've opted to cut out the concurrency entirely; with the rate limiting I've been seeing with even just two parallel workers, it's not really any faster than just one process making calls in a loop. Without a sizeable performance gain, I think it's a better idea to not have to deal with the complexities of channels and concurrent goroutines.

As for maybe trying to parallelize at the Docker level, we'd need to rewrite the tool a fair amount. Its current logic for discovering and gathering the task definitions to deregister would need to be removed, and the tool would instead need to accept pieces of work from an external operator. I guess in order to scale deregistration through Docker scaling, we would need to break up the tool into two Docker images - a worker image (this is the scalable one) that handles the logic of deregistering task definitions which have been handed to it, and a dispatcher image to gather all the task definitions, break them into chunks, and pass those chunks off to `n` containers running worker images. We'd need to have some sort of service discovery, or we'd need to make the dispatcher be configurable and in charge of spinning up the `n` worker containers. We'd also be replacing communication over go channels with communication between Docker containers.

File: cmd/ecs_task.go (hunk 2 above)
Comment (tlake) [reply cmd/ecs_task.go:62]: Not through testing, but with the removal of concurrency this shouldn't be a concern anymore.

File: ecsclient/ecsclient_test.go [hunk 3]
```diff
@@ -0,0 +1,689 @@
...
+			ServiceArns: []*string{aws.String("service2")},
+			NextToken:   nil,
+		}, nil)
+
+	result, err := e.CollectServices(clusterARNs)
+	if err != nil {
+		t.Error(err)
+	}
+
+	if equal := reflect.DeepEqual(expected, result); !equal {
```
Comment (diemonster) [inline ecsclient/ecsclient_test.go:112]: I was frequently told not to use reflection in tests in the past, so my knee-jerk reaction here was to say 'no'.... that being said, from the `DeepEqual` [docs](https://golang.org/pkg/reflect/#DeepEqual):

> Map values are deeply equal when all of the following are true: they are both nil or both non-nil, they have the same length, and either they are the same map object or their corresponding keys (matched using Go equality) map to deeply equal values.

This seems to be a lot more convenient than iterating through the maps and comparing values manually. Trying to see a downside to this but I can't rn.

**Review by diemonster — CHANGES_REQUESTED** [review]
Could you add a helm chart? https://docs.bitnami.com/kubernetes/how-to/create-your-first-helm-chart/

File: ecsclient/ecsclient_test.go (hunk 3 above)
Comment (tlake) [reply ecsclient/ecsclient_test.go]: Yeah, it seems pretty alright to me so far, so I dunno

Comment (tlake) [reply ecsclient/ecsclient_test.go:112]: Yeah, it seems pretty alright to me, soooooo /shrug

**Review by diemonster — APPROVED** [review]
LGTM!

**Pull request merged**